frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1658,6 +1658,13 @@ func dispatchCopy(d *dispatchState, cfg copyConfig) error {
 		fileOpt = append(fileOpt, llb.IgnoreCache)
 	}
 
+	if d.state.Output() == nil {
+		// Give a hint to DAP that this copy has no real parent.
+		fileOpt = append(fileOpt, llb.WithDescription(map[string]string{
+			"dap.hint.noparent": "true",
```

Comment:
I think we should be generic for the key and use the same format as https://github.com/moby/buildkit/blob/ee216b9a126bf939e119c89351bb084f77cc0aff/frontend/dockerfile/dockerfile2llb/convert.go#L1950-L1952

Maybe:

```suggestion
			"com.docker.dockerfile.v1.copy.noparent": "true",
```